components/object_store/src/obkv/mod.rs

```diff
@@ -0,0 +1,914 @@
+
+        // Before aborting multipart, we need to delete all data parts and meta info.
+        // Here to delete data with path `location` and multipart_id
+        let scan_context: ScanContext = ScanContext {
+            timeout: time::Duration::from_secs(meta::SCAN_TIMEOUT_SECS),
+            batch_size: meta::SCAN_BATCH_SIZE,
+        };
+
+        let prefix = format!("{location}@{multipart_id}@");
+        let scan_request = util::scan_request_with_prefix(prefix.as_bytes());
+
+        let mut iter = self
+            .client
+            .scan(scan_context, table_name, scan_request)
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+
+        let mut keys = vec![];
+        while iter.valid() {
+            keys.push(iter.key().to_vec());
+            iter.next().map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+        }
+
+        self.client
+            .batch_delete(table_name, keys)
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+
+        // Here to delete meta with path `location` and multipart_id
+        self.meta_manager
+            .delete_meta_with_version(location, multipart_id)
+            .await
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+
+        debug!(
+            "ObkvObjectStore abort_multipart operation, location:{location}, table_name:{table_name}, cost:{}",
+            instant.elapsed().as_millis()
+        );
+        Ok(())
+    }
+
+    async fn get(&self, location: &Path) -> Result<GetResult> {
+        let instant = Instant::now();
+        let result = self.get_internal(location).await;
+
+        debug!(
+            "ObkvObjectStore get operation, location:{location}, cost:{}",
+            instant.elapsed().as_millis()
+        );
+        result.box_err().map_err(|source| StoreError::NotFound {
+            path: location.to_string(),
+            source,
+        })
+    }
+
+    async fn get_range(&self, location: &Path, range: Range<usize>) -> Result<Bytes> {
+        let instant = Instant::now();
+
+        let table_name = self.pick_shard_table(location);
+        let meta =
+            self.read_meta(location)
+                .await
+                .box_err()
+                .map_err(|source| StoreError::NotFound {
+                    path: location.to_string(),
+                    source,
+                })?;
+
+        let batch_size = meta.part_size;
+        let key_list = meta.parts;
+        let start_index = range.start / batch_size;
+        let start_offset = range.start % batch_size;
+        let end_index = range.end / batch_size;
+        let end_offset = range.end % batch_size;
```

Comment:
Maybe we should define a method on meta to do the computing, and add some unit tests for it.

As for the method name, how about `compute_covered_parts`? And return a `CoveredParts` which should include the `part_keys` for the covered parts, `start_offset` and `end_offset`.

Reply:
done